hoot-core/src/main/cpp/hoot/core/io/OsmChangeWriter.cpp

```diff
@@ -40,56 +44,50 @@ void OsmChangeWriter::write(QIODevice &d, ChangeSetProviderPtr cs)
   writer.writeAttribute("version", "0.6");
   writer.writeAttribute("generator", "hootenanny");
 
-  Change::ChangeType last = Change::Unknown;
-
+  //Make the changeset have only one set of entries per change type.  Also for readability,
+  //sort the set of changes within each change type by element type.
+  QMap<Change::ChangeType, QMap<ElementType::Type, QVector<ConstElementPtr> > > elementsByChangesetTypeAndElementType;
```

Comment:
No, bad assumption on my part.  I see rails port tests where they are explicitly sending in multiple change docs of the same type within a changeset.  So, the extra sorting of changeset contents isn't needed.  I'll remove it and fix the test output.